From 0ca75f441714641a05983701e53cf8be7bf468ee Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbin.du@intel.com>
Date: Mon, 20 Apr 2015 17:45:13 +0800
Subject: [PATCH] usb/dwc3: invoke gadget reset callback on reset interrupt

Invoke gadget reset call back if it has when dwc receive
reset event.

Change-Id: I51ae7183006a9c05c1e2677594e52b2edc8adc46
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-30602
Signed-off-by: Du, Changbin <changbin.du@intel.com>
---
 drivers/usb/dwc3/gadget.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index b53c95464e10..1fcc493e0a62 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2358,6 +2358,12 @@ static void dwc3_gadget_reset_interrupt(struct dwc3 *dwc)
 	 */
 	if (dwc->ulpi_phy)
 		dwc3_gadget_kick_dog(dwc);
+
+	if (dwc->gadget_driver && dwc->gadget_driver->reset) {
+		spin_unlock(&dwc->lock);
+		dwc->gadget_driver->reset(&dwc->gadget);
+		spin_lock(&dwc->lock);
+	}
 }
 
 static void dwc3_update_ram_clk_sel(struct dwc3 *dwc, u32 speed)
-- 
1.9.1

